### DISCUSSION TITLE
string manipulation in pattern

### DISCUSSION THREAD

**Discussion by ahaberbosch** [faker-js/faker#2262] [Q&A] [answered]
Is there a way to do simple string manipulation in the patterns? e.g. replace(/'/g, '') I'm trying to avoid the use of apostrophes in faked street address, and the simplest way i've found to do it is by creating a custom locale and changing the street pattern to avoid using last name data, which includes names like "O'donnel", "O'Leary", etc.

**Comment by ST-DDT** [comment]
AFAICT there are only two ways two do this:

- create a new locale that overwrites the "bad" values
- access the data via rawDefinitions and overwrite them directly there

For the names or the addresses

**Reply by ahaberbosch** [reply]
Are there examples of doing either? I searched and searched and couldn't find them.

**Reply by matthewmayer** [reply]
i'd probably just write a simple wrapper
```js
function getFakeAddressWithoutApostrophe() {
   let address
   do {
      address = faker.location.street()
   } while (address.includes("'"))
   return address
}
```

**Reply by ST-DDT** [reply]
> Are there examples of doing either? I searched and searched and couldn't find them.

Creating a new Faker instance should be in the docs:

````ts
import { Faker, en } from "@faker-js/faker";

const { first_name: enFirstName, last_name: enLastName } = en.person;
const custom = {
  person: {
    first_name: enFirstName.map((v) => v.replace(/'/g, "")),
    last_name: enLastName.map((v) => v.replace(/'/g, "")),
  },
};

const customFaker = new Faker({ locale: [custom, en] });
````

----

Alternatively: (The following isn't officially supported usage:)

````ts
const rawPerson = faker.rawDefinitions.person;
rawPerson.first_name = rawPerson.first_name.map((v) => v.replace(/'/g, ""));
rawPerson.last_name = rawPerson.last_name.map((v) => v.replace(/'/g, ""));
````

**Reply by ahaberbosch** [reply, marked as answer]
I created a new class that extends Faker class
```typescript
import { Faker, en, base, en_US } from '@faker-js/faker'


class FilteredFaker extends Faker {
    constructor() {
        super({
            locale: [en_US, en, base],
        })
        this.rawDefinitions.person.last_name = this.rawDefinitions.person.last_name.filter(l => !l.includes(`'`))
    }
}

export const faker = new FilteredFaker()
```

It seems to work, please let me know if this has particular shortcomings
